presto-hive/src/main/java/io/prestosql/plugin/hive/HiveWriterFactory.java

```diff
@@ -605,9 +609,24 @@ public static String computeBucketedFileName(String queryId, int bucket)
     public static String getFileExtension(JobConf conf, StorageFormat storageFormat)
     {
         // text format files must have the correct extension when compressed
-        if (!HiveConf.getBoolVar(conf, COMPRESSRESULT) || !HiveIgnoreKeyTextOutputFormat.class.getName().equals(storageFormat.getOutputFormat())) {
+        if (!HiveConf.getBoolVar(conf, COMPRESSRESULT)) {
             return "";
         }
+        if (OrcOutputFormat.class.getName().equals(storageFormat.getOutputFormat())) {
```

Comment:
storageFormat.getOutputFormat() return value can be assigned to a local variable and use the same in if checks instead of calling this API multiple times.